py/nlr.h

```diff
@@ -64,8 +64,6 @@ struct _nlr_buf_t {
 #endif
 };
 
-extern nlr_buf_t *nlr_top;
```

Comment:
Since it's part of the global structure (btw what does ```ctx``` stand for?) you now also need something like

```
diff --git a/py/nlr.h b/py/nlr.h
index a8a8315..679cab8 100644
--- a/py/nlr.h
+++ b/py/nlr.h
@@ -68,8 +68,8 @@ struct _nlr_buf_t {
 NORETURN void nlr_setjmp_jump(void *val);
 // nlr_push() must be defined as a macro, because "The stack context will be
 // invalidated if the function which called setjmp() returns."
-#define nlr_push(buf) ((buf)->prev = nlr_top, nlr_top = (buf), setjmp((buf)->jmpbuf))
-#define nlr_pop() { nlr_top = nlr_top->prev; }
+#define nlr_push(buf) ((buf)->prev = MP_STATE_CTX(nlr_top), MP_STATE_CTX(nlr_top) = (buf), setjmp((buf)->jmpbuf))
+#define nlr_pop() { MP_STATE_CTX(nlr_top) = MP_STATE_CTX(nlr_top)->prev; }
 #define nlr_jump(val) nlr_setjmp_jump(val)
 #else
 unsigned int nlr_push(nlr_buf_t *);
diff --git a/py/nlrsetjmp.c b/py/nlrsetjmp.c
index 8b8e8d9..efd99ee 100644
--- a/py/nlrsetjmp.c
+++ b/py/nlrsetjmp.c
@@ -29,8 +29,8 @@
 #if MICROPY_NLR_SETJMP
 
 void nlr_setjmp_jump(void *val) {
-    nlr_buf_t *buf = nlr_top;
-    nlr_top = buf->prev;
+    nlr_buf_t *buf = MP_STATE_CTX(nlr_top);
+    MP_STATE_CTX(nlr_top) = buf->prev;
     buf->ret_val = val;
     longjmp(buf->jmpbuf, 1);
 }
```